src/pool/srv_pool.c

```diff
@@ -2847,11 +2852,20 @@ ds_pool_query_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		goto out_svc;
 
+	metrics = svc->ps_pool->sp_metrics[DAOS_POOL_MODULE];
+
 	/* See comment above, rebuild doesn't connect the pool */
 	if ((in->pqi_query_bits & DAOS_PO_QUERY_SPACE) &&
-	    !is_pool_from_srv(in->pqi_op.pi_uuid, in->pqi_op.pi_hdl))
+	    !is_pool_from_srv(in->pqi_op.pi_uuid, in->pqi_op.pi_hdl)) {
 		rc = pool_space_query_bcast(rpc->cr_ctx, svc, in->pqi_op.pi_hdl,
 					    &out->pqo_space);
+		if (unlikely(rc))
+			goto out_svc;
+
+		d_tm_inc_counter(metrics->query_space_total, 1);
+	} else {
+		d_tm_inc_counter(metrics->query_total, 1);
+	}
```

Comment:
Maybe I misunderstand what this function is doing, but should we always increment `metrics->query_total` and only conditionally increment `metrics->query_space_total`?

Reply:
+1 - I think we should increment query_total regardless.

Reply:
Good point.